shardingsphere-sql-parser/shardingsphere-sql-parser-dialect/shardingsphere-sql-parser-oracle/src/main/antlr4/imports/oracle/DMLStatement.g4

```diff
@@ -622,8 +622,34 @@ whereClause
     : WHERE expr
     ;
 
+hierarchicalQueryClause
+    : CONNECT BY NOCYCLE? expr (START WITH expr)?
+    | START WITH expr CONNECT BY NOCYCLE? expr
+    ;
+
 groupByClause
-    : GROUP BY orderByItem (COMMA_ orderByItem)*
+    : GROUP BY groupByItem (COMMA_ groupByItem)* havingClause?
+    ;
+
+groupByItem
+    : rollupCubeClause | groupingSetsClause | expr
+    ;
+
+rollupCubeClause
+    : (ROLLUP | CUBE) LP_ groupingExprList RP_
+    ;
+
+groupingSetsClause
+    : GROUPING SETS LP_ (rollupCubeClause | groupingExprList) (COMMA_ (rollupCubeClause | groupingExprList))* RP_
+    ;
+
+groupingExprList
+    : expressionList (COMMA_ expressionList)*
+    ;
+
+expressionList
+    : exprs
+    | LP_ expr? (COMMA_ expr?)* RP_
```

Comment:
Do you think this is better?
`exprs | LP_ expr? (COMMA_ expr)* RP_`

Reply:
Yes, I'll change that.